File: flare-android-client/flareclient-extensions-builtin/src/main/java/dev/flareframework/extensions/builtin/tasks/retry_connection/RetryConnectionTask.java
package dev.flareframework.extensions.builtin.tasks.retry_connection;

import android.app.Activity;
import dev.flareframework.client.FlareClientActivity;
import dev.flareframework.client.flare.task.FlareClientTask;
import org.json.JSONObject;

public class RetryConnectionTask implements FlareClientTask {
    public static final String ID = "retry_connection";

    @Override
    public String id() { return ID; }

    @Override
    public void execute(Activity host, JSONObject params) {
        if (host instanceof FlareClientActivity) {
            ((FlareClientActivity) host).retryConnection();
        }
    }
}

File: flare-web-client/src/flareclient-extensions-builtin/tasks/retry-connection/retry-connection-task.js
export const RetryConnectionTask = {
  id: "retry_connection",
  execute(_params) {
    if (window.__flareClient__) {
      window.__flareClient__.retryConnection();
    } else {
      console.warn("[RetryConnectionTask] no active FlareClient instance found");
    }
  }
};